packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -346,9 +346,9 @@ class _DropdownRoute<T> extends PopupRoute<_DropdownRouteResult<T>> {
     }
 
     if (scrollController == null) {
-      double scrollOffset = 0.0;
-      if (preferredMenuHeight > maxMenuHeight)
-        scrollOffset = selectedItemOffset - (buttonTop - menuTop);
+      final double scrollOffset = (preferredMenuHeight > maxMenuHeight) ?
+        (selectedItemOffset - (buttonTop - menuTop)).clamp(0.0, double.maxFinite)
```

Comment:
This is OK, but it seems like an odd way to say: 
`math.max(0.0, selectedItemOffset - (buttonTop - menuTop))`